StripePaymentSheet/StripePaymentSheet/Source/Helpers/DefaultPaymentMethodStore.swift

```diff
@@ -45,8 +46,20 @@ final class DefaultPaymentMethodStore {
         var customerToDefaultPaymentMethodID = UserDefaults.standard.customerToLastSelectedPaymentMethod ?? [:]
 
         let key = customerID ?? ""
+        // Set Default to legacy behavior
```

Comment:
I'm not sure what this is trying to do, as it's setting the field twice. Are we trying to migrate the default payment method option from the older style to this new JSON-encoded version? If so, I think it would make more sense to handle this on read:
* In `defaultPaymentMethod(for:)`, you could check for a modern value from `UserDefaults.standard.customerToLastSelectedPaymentMethod` (which would point to a new key for JSON-encoded selections).
* If one doesn't exist, you check `UserDefaults.standard.legacyCustomerToLastSelectedPaymentMethod` (which points to the old key) for a legacy value and convert it to the modern value.
* Otherwise, return nil.

Then in the `setDefaultPaymentMethod` code, you don't need to handle legacy values at all — just store the latest modern value.

Might be misunderstanding something though!

Reply:
i think this is just a misunderstanding -- I was thinking it was possible that
* `if let paymentMethodOption = PersistablePaymentMethodOption(legacyValue: identifier.value) {` - could fail, but unlikely, in which case, we wouldn't be writing anything to user defaults
* `try encoder.encode(...)` - could also fail, in which case, we are also not writing anything to user defaults.

Both of these cases are unlikely, but it was my attempt at writing "at least something" to user defaults. -- maybe we should just assume that both of those cases will succeed -- and if they fail, write nothing to user defaults?

Reply:
FWIW - i think `defaultPaymentMethod(for:)` is doing what you are outlining -- attempting to decode to the new version first, and if decoding fails, we call `return PaymentMethodIdentifier(value: value)` on line 92

Reply:
`encoder.encode` shouldn't fail, as we control the entire (very simple) struct — that's a programming error for us if it does. Seems okay to write nothing in that situation.

I'm confused as to why we're using `PersistablePaymentMethodOption(legacyValue:)` to construct this — why are we receiving a legacy value from a different part of our own SDK? Why wouldn't we initialize the `PersistablePaymentMethodOption` directly from the `PaymentMethodIdentifier`, instead of converting it to/from a raw value?

Reply:
I ran into an issue where `PersistablePaymentMethodOption` (which is a part of StripePaymentsUI) can't have an init which accepts a `PaymentMethodIdentifier` (which is a part of StripePaymentMethodSheet) .. but .. hmmm we could add an extension in `PaymentMethodIdentifier` to do this

Reply:
sorry for the churn here, ready for review!